src/rgw/rgw_rest_s3.cc

```diff
@@ -3521,31 +3542,30 @@ void RGWRestoreObj_ObjStore_S3::send_response()
     dump_start(s);
     return;
   }
-
-  rgw::sal::Attrs attrs = s->object->get_attrs();
-  auto attr_iter = attrs.find(RGW_ATTR_RESTORE_STATUS);
-  rgw::sal::RGWRestoreStatus restore_status;
-  if (attr_iter != attrs.end()) {
-    bufferlist bl = attr_iter->second;
-    auto iter = bl.cbegin();
-    decode(restore_status, iter);
-  }
-  ldpp_dout(this, 10) << "restore_status=" << restore_status << dendl;
   
-  if (attr_iter == attrs.end() || restore_status != rgw::sal::RGWRestoreStatus::None) {
-    s->err.http_ret = 202; //Accepted
-    dump_header(s, "x-amz-restore", rgw_bl_str(restore_status));
-  } else if (restore_status != rgw::sal::RGWRestoreStatus::RestoreAlreadyInProgress) {
-    s->err.http_ret = 409; // Conflict
-    dump_header_if_nonempty(s, "x-amz-restore", rgw_bl_str(restore_status));
-  } else if (restore_status != rgw::sal::RGWRestoreStatus::CloudRestored) {
-    s->err.http_ret = 200; // OK
-    dump_header_if_nonempty(s, "x-amz-restore", rgw_bl_str(restore_status));
-  } else {
-    s->err.http_ret = 202; // Accepted
-    dump_header_if_nonempty(s, "x-amz-restore", rgw_bl_str(restore_status));
-  }
+  auto attr_iter = attrs.find(RGW_ATTR_RESTORE_STATUS);
 
+  if (attr_iter == attrs.end() || restore_status == rgw::sal::RGWRestoreStatus::None) {
+    if (rt == rgw::sal::RGWRestoreType::Temporary) {
```

Comment:
if restore_status is None, why will RestoreType be set?

Reply:
Not have to fetch in the send response in the restore status.